src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs

```diff
@@ -53,7 +54,7 @@ internal sealed partial class ExpressionEval {
                     value = GetValueFromInstanceCall(pi, expr);
                     break;
                 case IPythonFunctionType ft: // Standalone function or a class method call.
-                    var instance = ft.DeclaringType != null ? new PythonInstance(ft.DeclaringType) : null;
+                    var instance = ft.DeclaringType != null ? ft.DeclaringType.CreateInstance(args) : null;
```

Comment:
`ft.DeclaringType?.CreateInstance(args)`, no need for conditional anymore.

Reply:
ok